GTFO-API/Utilities/LiveEdit.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GTFO.API.Utilities
+{
+    /// <summary>
+    /// LiveEditEventHandler
+    /// </summary>
+    /// <param name="e">EventArgs</param>
+    public delegate void LiveEditEventHandler(FileSystemEventArgs e);
+
+    /// <summary>
+    /// Utility class to make support of LiveEdit config file for plugins
+    /// </summary>
+    public static class LiveEdit
+    {
+        internal static readonly List<LiveEditListener> s_Listeners = new();
+
+        /// <summary>
+        /// Create the LiveEdit Listener and Allocate
+        /// </summary>
+        /// <param name="path">Base Path to Track the change</param>
+        /// <param name="filter">File filter to filter the files ie) *.json</param>
+        /// <param name="includeSubDir">Include Sub-directories?</param>
+        public static LiveEditListener CreateListener(string path, string filter, bool includeSubDir)
+        {
+            LiveEditListener listener = new(path, filter, includeSubDir)
+            {
+                m_Allocated = true
+            };
+
+            s_Listeners.Add(listener);
+            return listener;
+        }
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    [SuppressMessage("Interoperability", "CA1416:Platform Compatible on FileSystemWatcher", Justification = "GTFO is Windows only game")]
+    public sealed class LiveEditListener : IDisposable
+    {
+        /// <summary>
+        /// Event when File has Changed, Thread-safe
+        /// </summary>
+        public event LiveEditEventHandler FileChanged;
+        /// <summary>
+        /// Event when File has Deleted, Thread-safe
+        /// </summary>
+        public event LiveEditEventHandler FileDeleted;
+        /// <summary>
+        /// Event when File has Created, Thread-safe
+        /// </summary>
+        public event LiveEditEventHandler FileCreated;
+        /// <summary>
+        /// Event when File has Renamed, Thread-safe
+        /// </summary>
+        public event LiveEditEventHandler FileRenamed;
+
+        internal FileSystemWatcher m_Watcher = null;
+        internal bool m_Allocated = false;
```

Comment:
These don't need to be internal, especially while using the private member naming convention
```suggestion
        private FileSystemWatcher m_Watcher = null;
        private bool m_Allocated = false;
```